cmd/ko/commands.go

```diff
@@ -130,7 +130,20 @@ func addKubeCommands(topLevel *cobra.Command) {
 			if err != nil {
 				log.Fatalf("error piping to 'kubectl apply': %v", err)
 			}
-			go resolveFilesToWriter(fo, no, lo, stdin)
+
+			go func() {
+				// kubectl buffers data before starting to apply it, which
+				// can lead to resources being created more slowly than desired.
+				// In the case of --watch, it can lead to resources not being
+				// applied at all until enough iteration has occurred.  To work
+				// around this, we prime the stream with a bunch of empty objects
+				// which kubectl will discard.
```

Comment:
Maybe add this before and/or after the loop:

```go
stdin.Write([]byte("# See https://github.com/google/go-containerregistry/pull/348\n"))
```

For anyone who somehow stumbles across this?

Reply:
We only do this on the apply path, so opting for a comment in the code instead.